include/structured/Dialect/Indexing/IR/IndexingOps.td

```diff
@@ -65,4 +65,40 @@ def Indexing_ConcatenateOp : Indexing_Op<"concatenate",
   }];
 }
 
+def Indexing_ARangeOp : Indexing_Op<"arange", [
+    Pure,
+    DeclareOpInterfaceMethods<InferTypeOpInterface>,
+    SameVariadicOperandSize
+  ]> {
+  let summary = "Return evenly spaced values within a given interval.";
+  let description = [{
+    Example:
+    ```mlir
+    %range = arange(%start, %stop, %step) -> tensor<?xindex>
+    ```
+    Values are generated within the half-open interval [start, stop), with spacing between values given by step.
+  }];
+
+  let arguments = (ins Optional<I64>:$start,
+                       Optional<I64>:$stop,
+                       Optional<I64>:$step,
+                       OptionalAttr<I64Attr>:$startAttr,
+                       OptionalAttr<I64Attr>:$stopAttr,
+                       OptionalAttr<I64Attr>:$stepAttr);
+  let results = (outs 1DTensorOf<[Index]>:$result);
+  // the reason for the extra double backtick in (```start
+  // is because otherwise an extraneous space is emitted in the pretty print
```

Comment:
interesting, I had never realized this, thanks for your attention to detail!

Reply:
we all three can thank @zero9178 - he's the one that helped me fix on discord :)